pkg/vcs/vcs.go

```diff
@@ -151,28 +155,42 @@ type BisectEnv struct {
 	KernelConfig []byte
 }
 
+// NewRepo is intended for autonomous programs that fully manage the repo.
+// Such repo will try to re-create the checkout from scratch to recover from any errors,
```

Comment:
Looks like what you are after is "precious" vs. "discardable" repo, rather than ["not"] "user". Maybe adapt the terminology accordingly. Also would work better with enum.

Reply:
Renamed to precious and added typed options.